pkg/intel/metadata/manifest/key.go

```diff
@@ -131,19 +152,59 @@ func (k *Key) SetPubKey(key crypto.PublicKey) error {
 	return fmt.Errorf("unexpected key type: %T", key)
 }
 
-//PrintMEKey prints the KM public signing key hash to fuse into the Intel ME
-func (k *Key) PrintMEKey() error {
+//PrintBPMPubKey prints the BPM public signing key hash to fuse into the Intel ME
+func (k *Key) PrintBPMPubKey(bpmAlg Algorithm) error {
 	buf := new(bytes.Buffer)
 	if len(k.Data) > 1 {
-		if err := binary.Write(buf, binary.LittleEndian, k.Data[4:]); err != nil {
-			return nil
+		hash, err := bpmAlg.Hash()
+		if err != nil {
+			return err
+		}
+		if k.KeyAlg == AlgRSA {
+			if err := binary.Write(buf, binary.LittleEndian, k.Data[4:]); err != nil {
+				return err
+			}
+			hash.Write(buf.Bytes())
+			fmt.Printf("   Boot Policy Manifest Pubkey Hash: 0x%x\n", hash.Sum(nil))
+		} else if k.KeyAlg == AlgSM2 || k.KeyAlg == AlgECC {
+			if err := binary.Write(buf, binary.LittleEndian, k.Data); err != nil {
+				return err
+			}
+			hash.Write(buf.Bytes())
+			fmt.Printf("   Boot Policy Manifest Pubkey Hash: 0x%x\n", hash.Sum(nil))
+		} else {
+			fmt.Printf("   Boot Policy Manifest Pubkey Hash: Unknown Algorithm\n")
 		}
-		if err := binary.Write(buf, binary.LittleEndian, k.Data[:4]); err != nil {
-			return nil
+	} else {
+		fmt.Printf("   Boot Policy Pubkey Hash: No km public key set in KM\n")
+	}
+
+	return nil
+}
+
+//PrintKMPubKey prints the KM public signing key hash to fuse into the Intel ME
+func (k *Key) PrintKMPubKey(kmAlg Algorithm) error {
+	buf := new(bytes.Buffer)
+	if len(k.Data) > 1 {
+		if k.KeyAlg == AlgRSA {
+			if err := binary.Write(buf, binary.LittleEndian, k.Data[4:]); err != nil {
+				return err
+			}
+			if err := binary.Write(buf, binary.LittleEndian, k.Data[:4]); err != nil {
+				return err
+			}
+			if kmAlg != AlgSHA256 {
+				return fmt.Errorf("KM public key hash algorithm must be SHA256")
+			}
+			hash, err := kmAlg.Hash()
```

Comment:
Is there an indicator in spec, that the hash function can be SHA384 or SM3?
If so:

```
if kmAlg == AlgSHA256 || kmAlg == AlgSHA384 || kmAlg == AlgSM3 {
    hash, err := kmAlg.Hash()
    if err != nil {
        return err 
    }
} else {
	return fmt.Errorf("KM public key hash algorithm must be SHA256")
}
hash.Write(buf.Bytes())
``

Reply:
KM pubkey digest can be either SHA256 or SM3 to fit into 32 bytes of FPFs. SHA1
digest despite that it fits into 32 bytes is not supported since it was historically
forbidden in HSW and SKL platforms.
From the BIOS Specification #575623